docs/src/modules/components/AppWrapper.js

```diff
@@ -109,86 +103,33 @@ async function registerServiceWorker() {
   }
 }
 
-class AppWrapper extends React.Component {
-  state = {};
+const useEnhancedEffect = typeof window === 'undefined' ? React.useEffect : React.useLayoutEffect;
 
-  componentDidMount() {
-    themeSideEffect(this.props.reduxTheme);
+function AppWrapper(props) {
+  const { children } = props;
 
+  useEnhancedEffect(() => {
```

Comment:
We are using `useLayout()` in the demos (next & ssr). What do you think of following them? Why?
I don't think that we need to block the main thread to remove the server side generated styles. We reduce the pressure on the main thread. This way, we could merge it with the hook for `registerServiceWorker()`.

Reply:
> We are using useLayout() in the examples (next & ssr).

You mean `useEffect`? `useLayoutEffect` would be equivalent. `useEffect` will cause styles collision between first commit and effect phase. This should be noticeable if you use a non-default theme.

I thought I tested this initially and it didn't look good. Though this was with the theme in redux. Not sure if this was related or if I just looked in the wrong place.

In any case: `useEffect` works too and if `useEffect` works then it is always superior to `useLayoutEffect`.